src/utils.ts

```diff
@@ -14,6 +16,10 @@ export const normalizeKeys = (keyOrKeys: KeyOrKeys): string[] => (
     map(Array.isArray(keyOrKeys) ? keyOrKeys : [keyOrKeys], normalizeKey)
 );
 
+export const formCompoundKey = (...keys: (string | number)[]): string => {
+    return keys.map(k => k.toString()).join(CompoundKeyJoinerString);
+};
+
```

Comment:
nit: `keys` can be `string` or `number`, `.join` converts both versions to `string`

```javascript
[1, 2].map(k => k.toString()).join('%&'); => "1%&2"
['1', '2'].map(k => k.toString()).join('%&'); => "1%&2" 

[1, 2].join('%&'); => "1%&2"
['1', '2'].join('%&'); => "1%&2"

```
Is there a special case which requires to use `.map`?

Reply:
Oh, interesting, I did not actually know that nuance of the join function.  I figured I had to turn it into a string array first.  Too used to more strongly typed compiled languages. :D